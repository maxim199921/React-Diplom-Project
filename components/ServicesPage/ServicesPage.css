.ServicesPage {
  margin-top: 120px;
  background-color: #efefef;
}

.ServicesPage-wrapper {
  margin-bottom: 50px;
  background-color: #5dbbc1;
  color: white;
  box-shadow: 0 10px 15px 0 rgba(22,35,23,.7);
}

.ServicesPage-services-text,
.ServicesPage-Comments-text{
  padding-top: 15px;
  padding-bottom: 15px;
  font-size: 36px;
  line-height: 40px;
  font-weight: bold;
  text-align: center;
}

.HomePage-Services-text {
  font-weight: bold;
  font-size: 20px;
}