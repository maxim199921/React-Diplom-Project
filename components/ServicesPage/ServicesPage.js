import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";

import ServicesComments from './ServicesComments/ServicesComments'
import './ServicesPage.css';
import {loadDataComments} from '../../redux/ActionCreaters/CommentsActions'
import {returnSwitchLastArr} from '../../modules/modules'


class ServicesPage extends React.PureComponent {

  static propTypes = {
      commentsList: PropTypes.arrayOf(
          PropTypes.shape({
              name: PropTypes.string.isRequired,
              comment: PropTypes.string.isRequired,
          })
      ),
  };

    componentDidMount = () => {
        this.props.dispatch(loadDataComments());
    };

  render() {

      let commentList = this.props.commentsList.map((item, index) =>
          <ServicesComments key={index}
                      info={item}/>
      );

    return (
      <div className="ServicesPage">
          <div className="container">
            <div className="row">
                <div className="col-md-6">
                    <div className="row">
                        <div className="col-md-12 ServicesPage-wrapper">
                            <h2 className="ServicesPage-Comments-text">Comments</h2>
                        </div>
                        <div className="col-md-12">
                            <div className="row">
                                {commentList}
                            </div>
                        </div>
                    </div>
                </div>
                <div className="col-md-6">
                    <div className="row">
                        <div className="col-md-12 ServicesPage-wrapper">
                            <h2 className="ServicesPage-services-text">Services</h2>
                        </div>
                        <div className="col-md-12">
                            <div className="row">
                                <div className="col-12 col-sm-6">
                                    <div className="row">
                                        <div className="col-12 HomePage-Services-img-wrapper">
                                            <div className="HomePage-Services-img1">
                                                <span hidden={true}>img</span>
                                            </div>
                                        </div>
                                        <div className="col-12">
                                            <p className="HomePage-Services-text">Work with love</p>
                                        </div>
                                    </div>
                                </div>
                                <div className="col-12 col-sm-6">
                                    <div className="row">
                                        <div className="col-12 HomePage-Services-img-wrapper">
                                            <div className="HomePage-Services-img2">
                                                <span hidden={true}>img</span>
                                            </div>
                                        </div>
                                        <div className="col-12">
                                            <p className="HomePage-Services-text">Best guarantee</p>
                                        </div>
                                    </div>
                                </div>
                                <div className="col-12 col-sm-6">
                                    <div className="row">
                                        <div className="col-12 HomePage-Services-img-wrapper">
                                            <div className="HomePage-Services-img3">
                                                <span hidden={true}>img</span>
                                            </div>
                                        </div>
                                        <div className="col-12">
                                            <p className="HomePage-Services-text">Permanent discount</p>
                                        </div>
                                    </div>
                                </div>
                                <div className="col-12 col-sm-6">
                                    <div className="row">
                                        <div className="col-12 HomePage-Services-img-wrapper">
                                            <div className="HomePage-Services-img4">
                                                <span hidden={true}>img</span>
                                            </div>
                                        </div>
                                        <div className="col-12">
                                            <p className="HomePage-Services-text">Fast delivery</p>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
          </div>
      </div>
    );

  }

}


const mapStateToProps = function (state) {
    return {
        commentsList: returnSwitchLastArr(state.comments.commentsList, -5),
    };
};


export default connect(mapStateToProps)(ServicesPage);
