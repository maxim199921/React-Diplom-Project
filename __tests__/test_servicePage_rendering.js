import React from 'react';
import renderer from 'react-test-renderer';
import ServicesPage from '../components/ServicesPage/ServicesPage';
import {HashRouter, hashHistory} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from '../redux/configureStore';
const store = configureStore({});

test('тест на рендеринг сктраницы сервисов', () => {

    const component = renderer.create(
        <Provider store={store}>
            <HashRouter history={hashHistory}>
                <ServicesPage/>
            </HashRouter>
        </Provider>
    );

    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});