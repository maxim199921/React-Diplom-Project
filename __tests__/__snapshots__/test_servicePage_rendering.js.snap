// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тест на рендеринг сктраницы сервисов 1`] = `
<div
  className="ServicesPage"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12 ServicesPage-wrapper"
          >
            <h2
              className="ServicesPage-Comments-text"
            >
              Comments
            </h2>
          </div>
          <div
            className="col-md-12"
          >
            <div
              className="row"
            />
          </div>
        </div>
      </div>
      <div
        className="col-md-6"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12 ServicesPage-wrapper"
          >
            <h2
              className="ServicesPage-services-text"
            >
              Services
            </h2>
          </div>
          <div
            className="col-md-12"
          >
            <div
              className="row"
            >
              <div
                className="col-12 col-sm-6"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 HomePage-Services-img-wrapper"
                  >
                    <div
                      className="HomePage-Services-img1"
                    >
                      <span
                        hidden={true}
                      >
                        img
                      </span>
                    </div>
                  </div>
                  <div
                    className="col-12"
                  >
                    <p
                      className="HomePage-Services-text"
                    >
                      Work with love
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-12 col-sm-6"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 HomePage-Services-img-wrapper"
                  >
                    <div
                      className="HomePage-Services-img2"
                    >
                      <span
                        hidden={true}
                      >
                        img
                      </span>
                    </div>
                  </div>
                  <div
                    className="col-12"
                  >
                    <p
                      className="HomePage-Services-text"
                    >
                      Best guarantee
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-12 col-sm-6"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 HomePage-Services-img-wrapper"
                  >
                    <div
                      className="HomePage-Services-img3"
                    >
                      <span
                        hidden={true}
                      >
                        img
                      </span>
                    </div>
                  </div>
                  <div
                    className="col-12"
                  >
                    <p
                      className="HomePage-Services-text"
                    >
                      Permanent discount
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-12 col-sm-6"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 HomePage-Services-img-wrapper"
                  >
                    <div
                      className="HomePage-Services-img4"
                    >
                      <span
                        hidden={true}
                      >
                        img
                      </span>
                    </div>
                  </div>
                  <div
                    className="col-12"
                  >
                    <p
                      className="HomePage-Services-text"
                    >
                      Fast delivery
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
